lang/cs/CoreCLRCompatibility/AnalyzeCoreCLRCompatibility.ps1

```
@@ -0,0 +1,124 @@
+$exeDotNetTargets = @(".NET Core + Platform Extensions, Version=1.0", ".NET Framework, Version=4.5.1")
+
+# Set up the directory structure
+If (Test-Path $outputDirectory)
+{
+	Remove-Item $outputDirectory -Force -Recurse
+}
+New-Item -ItemType Directory -Force -Path $outputDirectory | Out-Null
+
+# Install the API Portability Analyzer if it is not present
+$portabilityAnalyzer = ".\ApiPort\ApiPort.exe"
+If (!(Test-Path $portabilityAnalyzer))
+{
+	$analyzerUrl="https://github.com/Microsoft/dotnet-apiport/releases/download/v1.4.0.alpha.00175/ApiPort.zip"
+	$analyzerZip="$(pwd)\analyzer.zip"
+	wget $analyzerUrl -OutFile $analyzerZip
+	Add-Type -assembly "System.IO.Compression.Filesystem";
+	[IO.Compression.Zipfile]::ExtractToDirectory($analyzerZip, (pwd));
+	rm $analyzerZip
+}
+
+# Set up the counter
+[int] $totalIncompatibilities = 0
+
+# Start a unique list of dependencies
+$missingDependencies = @{}
+
+# Get all the folders
+Get-ChildItem $baseDir -dir -Filter "Org.Apache.*" |
+	ForEach-Object {
+		$projectName = $_.Name
+
+		# Verify that the project is part of the .NET solution
+		$csProj = "${baseDir}\${projectName}\${projectName}.csproj"
+		If (!(Test-Path $csProj))
+		{
+			echo "Skipping ${projectName}: No csproj"
+			return
+		}
+
+		# Determine the output type for the project
+		$outputType = $(Select-String -Path $csProj -Pattern "<OutputType>").Line -replace '^\s+<OutputType>(.+?)</OutputType>\s*$','$1'
+
+		# Get the portability settings for this project
+		$objectSuffix = ""
+		$dotNetTargets = @()
+		If ($outputType -eq "Library")
+		{
+			$objectSuffix = "dll"
+			$dotNetTargets = $libraryDotNetTargets
+		}
+		ElseIf ($outputType -eq "Exe")
+		{
+			$objectSuffix = "exe"
+			$dotNetTargets = $exeDotNetTargets
+		}
+		Else
+		{
+			echo "Skipping ${projectName}: OutputType is not 'Library','Exe'"
+			return
+		}
+
+		# Define the object we want to explore
+		$objectPath = "${binDir}\${projectName}\${projectName}.${objectSuffix}"
```

This this fail explicitly when the file isn't found?